packages/lesswrong/server/editor/make_editable_callbacks.js

```diff
@@ -24,39 +25,48 @@ function mjPagePromise(html, beforeSerializationCallback) {
   })
 }
 
-const createHtmlHighlight = (body) => {
-  if (body.length > 2400) {
-    // drop the last paragraph
-    const highlight2400Shortened = body.slice(0,2400).split("\n").slice(0,-1).join("\n")
-    const highlightnewlineShortened = body.split("\n\n").slice(0,5).join("\n\n")
-    if (highlightnewlineShortened.length > highlight2400Shortened.length) {
-      return mdi.render(highlight2400Shortened)
-    } else {
-      return mdi.render(highlightnewlineShortened)
-    }
-  } else {
-    return mdi.render(body)
+export const getExcerptFieldsFromMarkdown = (markdownBody, fieldName, mdi) => {
```

Comment:
Refactor this to use `getExcerptFieldsFromHTML` or the other way around, since they share 90% of the code.